### PR TITLE
Make version_defaults.mk findable by using correct $ANDROID_ROOT base

### PR DESCRIPTION
**Pull request by pablomh**
Appease shellcheck while at it.

### CODE REVIEW

File: create_images.sh [hunk 1]
```diff
@@ -15,35 +15,35 @@ echo "==> Generating images for patch level $_os_patch_level"
 
 # mkdir -p $(dirname $_boot_out)
 
-if [ "$_has_dtbo" = "true" ]; then
-    _dts_folder=$(realpath $_out/arch/arm64/boot/dts/qcom)
-    _files=$(find $_dts_folder -iname "*.dtbo")
+if [[ "$_has_dtbo" == "true" ]]; then
+    _dts_folder=$(realpath "$_out"/arch/arm64/boot/dts/qcom)
```
Comment (MarijnS95) [inline create_images.sh]: I personally find this more readable:

```suggestion
    _dts_folder=$(realpath "$_out/arch/arm64/boot/dts/qcom")
```

Would you mind applying that to all cases?

File: create_images.sh [hunk 2]
```diff
@@ -15,35 +15,35 @@ echo "==> Generating images for patch level $_os_patch_level"
...
+    _files=$(find "$_dts_folder" -iname "*.dtbo")
     echo "==> Creating dtboimg from $_files"
-    python2 $_kernel_path/scripts/mkdtboimg.py create $_device-dtbo.img --page_size=$BOARD_KERNEL_PAGESIZE $_files
+    python2 "$_kernel_path"/scripts/mkdtboimg.py create "$_device-dtbo".img --page_size="$BOARD_KERNEL_PAGESIZE" "$_files"
```
Comment (MarijnS95) [inline create_images.sh]:
```suggestion
    python2 "$_kernel_path/scripts/mkdtboimg.py" create "$_device-dtbo.img" --page_size="$BOARD_KERNEL_PAGESIZE" "$_files"
```

(Note that `-dtbo` wasn't part of the expansion, this one should be read as `${_device}-dtbo.img`)

Perhaps we should see if this script can finally handle `python3` too and/or has a `/usr/bin/env` shebang.

**Review by MarijnS95 — APPROVED** [review]
Thanks for clearing up even more missing `$ANDROID_ROOT` cases! Would you mind adjusting quoting positions though? I find it more clear to put the entire string in quotes rather than just the macro expansions. This is closer to format strings in most programming languages too.

**Comment by MarijnS95** [comment]
Now that we're at the subject of quoting, I pushed https://github.com/MarijnS95/oot/commit/f898f28195008476f0daf05e718e137de0253522 recently but the nested quotes aren't "taken care of" by make, resuling in a:

```
/data/aosp-q/kernel/sony/msm-4.14/kernel/scripts/link-vmlinux.sh: line 61: "/data/gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-"ar: No such file or directory
```

Do you happen to know how to properly deal with these quoting rules, allowing a path with spaces in `_cross_compile{,_32}`?

**Comment by pablomh** [comment]
Re-reading what https://github.com/koalaman/shellcheck/wiki/SC2086 has to say about it, I agree with you and will reformat accordingly :)

**Comment by MarijnS95** [comment]
Neat, canonical quoting is the name!

**Pull request merged**